src/kern/npf_alg_pptp.c

```diff
@@ -515,17 +523,13 @@ pptp_tcp_translate(npf_cache_t *npc, npf_nat_t *nt, bool forw)
 		if (pptp->len < sizeof(pptp_outgoing_call_reply_t)) {
 			return false;
 		}
-		pptp_call_reply = nbuf_ensure_contig(nbuf,
-		    sizeof(pptp_outgoing_call_reply_t));
-		if (!pptp_call_reply) {
-			return false;
-		}
+		pptp_call_reply = (pptp_outgoing_call_reply_t *)pptp;
 
 		/* Lookup the GRE connection context. */
 		mutex_enter(&tcp_ctx->lock);
 		gre_state = pptp_gre_lookup_state(tcp_ctx,
 		    CLIENT_CALL_ID, pptp_call_reply->peer_call_id);
-		if (gre_state == NULL || gre_state->call_id[SERVER_CALL_ID] != 0) {
+		if (gre_state == NULL || (gre_state->flags & GRE_STATE_SERVER_CALL_ID)) {
```

Comment:
So, can call ID be zero? Would it be a regulal value or some reserved value?

Reply:
I am not sure does RFC allow using zero values, but some pptp clients use zero values for sure. And there was complains from my clients so I started to investigate.

Reply:
- Is `0xffff` a valid call ID too?
- Styling: please check flags against zero as follows:
```c
(gre_state->flags & GRE_STATE_SERVER_CALL_ID) != 0
```

Reply:
yes, I think 0xffff is a valid call ID.